crates/red_knot_python_semantic/src/semantic_index/ast_ids.rs

```diff
@@ -207,4 +219,10 @@ pub(crate) mod node_key {
             Self(NodeKey::from_node(value))
         }
     }
+
+    impl From<&ast::Identifier> for ExpressionNodeKey {
+        fn from(value: &ast::Identifier) -> Self {
+            Self(NodeKey::from_node(value))
+        }
+    }
```

Comment:
An `ast::Identifier` may be an AST node, but it definitely isn't an expression, so I'm not sure if this is really correct. `except` handlers don't use `ast::Name` nodes for the symbols they bound, however, so I'm not sure how to do this otherwise.